pkg/network/infraconfigurators/bridge_test.go

```diff
@@ -189,4 +189,621 @@ var _ = Describe("Bridge infrastructure configurator", func() {
+				handler.EXPECT().CreateTapDevice(tapDeviceName, queueCount, launcherPID, mtu, netdriver.LibvirtUserAndGroupId)
+				handler.EXPECT().BindTapDeviceToBridge(tapDeviceName, bridgeName)
+			}
+		}
+
+		withErrorCreatingTapDevice := func(tapDeviceName string, errorString string) Option {
+			return func(handler *netdriver.MockNetworkHandler) {
+				handler.EXPECT().CreateTapDevice(
+					tapDeviceName, queueCount, launcherPID, mtu, netdriver.LibvirtUserAndGroupId).Return(
+					fmt.Errorf(errorString))
+			}
+		}
+
+		withDisabledTxOffloadChecksum := func(bridgeName string) Option {
+			return func(handler *netdriver.MockNetworkHandler) {
+				handler.EXPECT().DisableTXOffloadChecksum(bridgeName)
+			}
+		}
+
+		withLinkLearningOff := func(link netlink.Link) Option {
+			return func(handler *netdriver.MockNetworkHandler) {
+				handler.EXPECT().LinkSetLearningOff(link)
+			}
+		}
+
+		withErrorSettingPodLinkUp := func(link netlink.Link, errorString string) Option {
+			return func(handler *netdriver.MockNetworkHandler) {
+				handler.EXPECT().LinkSetUp(link).Return(fmt.Errorf(errorString))
+			}
+		}
+
+		withErrorSettingLinkLearningOff := func(link netlink.Link, errorString string) Option {
+			return func(handler *netdriver.MockNetworkHandler) {
+				handler.EXPECT().LinkSetLearningOff(link).Return(fmt.Errorf(errorString))
+			}
+		}
+
+		When("the pod features an L3 configured network (IPAM)", func() {
+			var (
+				dummySwap              *netlink.Dummy
+				podLinkAfterNameChange *netlink.GenericLink
+			)
+
+			BeforeEach(func() {
+				bridgeIPStr = fmt.Sprintf(bridgeFakeIP, 0)
+				bridgeIPAddr, _ = netlink.ParseAddr(bridgeIPStr)
+				dummySwap = &netlink.Dummy{LinkAttrs: netlink.LinkAttrs{Name: ifaceName}}
+				inPodBridge = &netlink.Bridge{LinkAttrs: netlink.LinkAttrs{Name: bridgeIfaceName}}
+				podIP = netlink.Addr{IPNet: &net.IPNet{IP: net.IPv4(10, 35, 0, 6), Mask: net.CIDRMask(24, 32)}}
+				podLinkAfterNameChange = &netlink.GenericLink{LinkAttrs: netlink.LinkAttrs{Name: generateDummyIfaceName(ifaceName), MTU: mtu}}
+			})
+
+			withOriginalPodLinkDown := func(link netlink.Link) Option {
+				return func(handler *netdriver.MockNetworkHandler) {
+					handler.EXPECT().LinkSetDown(podLink)
+				}
+			}
+
+			withSwitcharoo := func(oldPodLink netlink.Link, renamedPodLink netlink.Link, newDummy netlink.Link, ip netlink.Addr) Option {
+				return func(handler *netdriver.MockNetworkHandler) {
+					handler.EXPECT().LinkAdd(newDummy)
+					handler.EXPECT().AddrReplace(newDummy, &ip)
+					handler.EXPECT().AddrDel(oldPodLink, &ip)
+					handler.EXPECT().LinkSetName(oldPodLink, renamedPodLink.Attrs().Name)
+					handler.EXPECT().LinkByName(renamedPodLink.Attrs().Name).Return(podLinkAfterNameChange, nil)
+				}
+			}
+
+			withPodLinkRandomMac := func(link netlink.Link, mac net.HardwareAddr) Option {
+				return func(handler *netdriver.MockNetworkHandler) {
+					handler.EXPECT().SetRandomMac(link.Attrs().Name).Return(mac, nil)
```

Comment:
Here the argument `mac` is shadowing the external `mac`, which is fine... but very easy to miss.